homeassistant/components/arlo/sensor.py

```diff
@@ -51,43 +82,44 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     sensors = []
     for sensor_type in config[CONF_MONITORED_CONDITIONS]:
+        sensor_entry = SensorEntityDescription(SENSOR_TYPES[sensor_type])
         if sensor_type == "total_cameras":
-            sensors.append(ArloSensor(SENSOR_TYPES[sensor_type][0], arlo, sensor_type))
+            sensors.append(ArloSensor(sensor_entry.name, arlo, sensor_entry))
         else:
             for camera in arlo.cameras:
                 if sensor_type in ("temperature", "humidity", "air_quality"):
                     continue
 
-                name = f"{SENSOR_TYPES[sensor_type][0]} {camera.name}"
-                sensors.append(ArloSensor(name, camera, sensor_type))
+                name = f"{sensor_entry.name} {camera.name}"
+                sensors.append(ArloSensor(name, camera, sensor_entry))
 
             for base_station in arlo.base_stations:
                 if (
                     sensor_type in ("temperature", "humidity", "air_quality")
                     and base_station.model_id == "ABC1000"
                 ):
-                    name = f"{SENSOR_TYPES[sensor_type][0]} {base_station.name}"
-                    sensors.append(ArloSensor(name, base_station, sensor_type))
+                    name = f"{sensor_entry.name} {base_station.name}"
+                    sensors.append(ArloSensor(name, base_station, sensor_entry))
 
     add_entities(sensors, True)
 
 
 class ArloSensor(SensorEntity):
     """An implementation of a Netgear Arlo IP sensor."""
 
-    def __init__(self, name, device, sensor_type):
+    def __init__(self, name, device, sensor_entry):
         """Initialize an Arlo sensor."""
         _LOGGER.debug("ArloSensor created for %s", name)
-        self._name = name
+        sensor_entry.name = name
+        if sensor_entry.key == "temperature":
+            sensor_entry.device_class = DEVICE_CLASS_TEMPERATURE
+        elif sensor_entry.key == "humidity":
+            sensor_entry.device_class = DEVICE_CLASS_HUMIDITY
+        self.entity_description = sensor_entry
         self._data = device
-        self._sensor_type = sensor_type
+        self._sensor_type = sensor_entry.key
         self._state = None
-        self._icon = f"mdi:{SENSOR_TYPES.get(self._sensor_type)[2]}"
-
-    @property
-    def name(self):
-        """Return the name of this camera."""
-        return self._name
+        self._icon = f"mdi:{sensor_entry.icon}"
```

Comment:
Add this prefix to the `SensorEntityDescription`  keys instead

Reply:
Done

Reply:
I see I misinterpreted this one, you talk about the "mdi:", corrected

> It looks like you might be missing a commit, some of my earlier suggestions has been resuggested

I can see why you think that, truth however is that I misunderstood what you wanted, but it is now corrected.

Your suggestion with next() in test_sensor.py is really elegant, thanks for that idea, that is something I will play with and use in the future.